coral/src/app/pages/index.test.tsx

```diff
@@ -1,15 +1,58 @@
 import HomePage from "src/app/pages";
 import { render, cleanup, screen } from "@testing-library/react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactElement } from "react";
+
+export function renderWithClient(client: QueryClient, ui: ReactElement) {
+  const { rerender, ...result } = render(
+    <QueryClientProvider client={client}>{ui}</QueryClientProvider>
+  );
+  return {
```

Comment:
If all possible, I think this utility should not return a value and everything should be accessed through `screen`. Also, the return value of this function is not used in the tests.

Reply:
Also very good point!